plugin/src/main/kotlin/com/vanniktech/maven/publish/Platform.kt

```diff
@@ -493,13 +492,13 @@ private fun setupTestFixtures(project: Project, sourcesJar: Boolean) {
       val action = Action<JvmVariantBuilder> {
         it.withSourcesJar().published()
       }
-      if (GradleVersion.current() >= GradleVersion.version("8.1-rc-1")) {
+      if (GradleVersion.current() >= GradleVersion.version("8.1")) {
         val extension = project.extensions.getByType(JavaPluginExtension::class.java)
         val testFixturesSourceSet = extension.sourceSets.maybeCreate(variant)
-        val method = services.javaClass.getMethod("createJvmVariant", String::class.java, SourceSet::class.java, Action::class.java)
-        method.invoke(services, variant, testFixturesSourceSet, action)
+        services.createJvmVariant(variant, testFixturesSourceSet, action)
       } else {
-        project.serviceOf<JvmModelingServices>().createJvmVariant(variant, action)
+        val method = services.javaClass.getMethod("createJvmVariant", String::class.java, Action::class.java)
+        method.invoke(services, variant, action)
       }
```

Comment:
Just swapped which version does the direct method call and which uses reflection